### PR TITLE
Array setting

### PR DESCRIPTION
**Pull request by TaroEld**
Create new array setting.

### CODE REVIEW

File: msu/systems/mod_settings/elements/array_setting.nut [hunk 1]
```diff
@@ -0,0 +1,44 @@
+::MSU.Class.ArraySetting <- class extends ::MSU.Class.AbstractSetting
+{
+	static Type = "Array";
+
+	constructor( _id, _value, _name = null, _description = null, _lockLength = false)
+	{
+		::MSU.requireArray( _value);
+		for (local i = 0; i < _value.len(); ++i)
+		{
+			::MSU.requireOneFromTypes(["string", "integer", "float", "array"], _value[i]);
```
Comment (Enduriel) [inline msu/systems/mod_settings/elements/array_setting.nut:10]: What happens if it's an array?

File: msu/systems/mod_settings/elements/array_setting.nut [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		this.Data.lockLength <- _lockLength;
+	}
+
+	function getUIData(_flags = [])
+	{
+		local ret = base.getUIData(_flags);
+		foreach(key, value in ret.value)
+		{
+			this.logInfo(value[0] + " " + value[1])
+		}
```
Comment (Enduriel) [inline msu/systems/mod_settings/elements/array_setting.nut:31]: leftover logging

File: ui/mods/msu/css/settings_screen.css [hunk 3]
```diff
@@ -241,52 +245,76 @@
 	width: 60.0rem;
 	height: 30.0rem;
 
-	background-image: url("coui://gfx/ui/skin/popup_background_600x300.png");
+	background-image: url("coui://gfx/mods/skin/generic_background_large_border.png");
```
Comment (Enduriel) [inline ui/mods/msu/css/settings_screen.css:248]: mods/msu, and we should convert our other stuff to be the same

**Review by Enduriel — CHANGES_REQUESTED** [review]

File: msu/systems/mod_settings/elements/array_setting.nut (hunk 1 above)
Comment (TaroEld) [reply msu/systems/mod_settings/elements/array_setting.nut:10]: Added some input validation

File: ui/mods/msu/css/settings_screen.css (hunk 3 above)
Comment (TaroEld) [reply ui/mods/msu/css/settings_screen.css]: fair

File: msu/systems/mod_settings/elements/array_setting.nut (hunk 2 above)
Comment (TaroEld) [reply msu/systems/mod_settings/elements/array_setting.nut:38]: thanks

**Review by Enduriel — APPROVED** [review]

**Pull request merged**